test/unit/assorted/server_selection_latency_window_utils.ts

```diff
@@ -0,0 +1,150 @@
+import { EJSON } from 'bson';
+import { expect } from 'chai';
+import { readdirSync, readFileSync } from 'fs';
+import { join } from 'path';
+import { promisify } from 'util';
+
+import { ReadPreference } from '../../../src/read_preference';
+import { ServerType, STATE_CONNECTED, TopologyType } from '../../../src/sdam/common';
+import { Server } from '../../../src/sdam/server';
+import { Topology } from '../../../src/sdam/topology';
+import { serverDescriptionFromDefinition } from './server_selection_spec_helper';
+
+interface ServerSelectionLatencyWindowTest {
+  description: string;
+  topology_description: {
+    type: TopologyType;
+    servers: ReadonlyArray<{ address: string; avg_rtt_ms: number; type: ServerType }>;
+  };
+
+  mocked_topology_state: ReadonlyArray<{
+    address: string;
+    operation_count: number;
+  }>;
+
+  iterations: number;
+
+  outcome: {
+    tolerance: number;
+    expected_frequencies: {
+      [key: string]: number;
+    };
+  };
+}
+
+type Outcome = ServerSelectionLatencyWindowTest['outcome'];
+type FrequencyMap = Outcome['expected_frequencies'];
+
+export function loadLatencyWindowTests(directory: string) {
+  const files = readdirSync(directory).filter(fileName => fileName.endsWith('.json'));
+
+  const tests: ServerSelectionLatencyWindowTest[] = [];
+
+  for (const fileName of files) {
+    const path = join(directory, fileName);
+    const contents = readFileSync(path, { encoding: 'utf-8' });
+    tests.push(EJSON.parse(contents) as ServerSelectionLatencyWindowTest);
+  }
+
+  return tests;
+}
+
+function compareResultsToExpected(
+  { tolerance, expected_frequencies }: Outcome,
+  observed_frequencies: FrequencyMap
+) {
+  const expectedFrequencies = Object.entries(expected_frequencies);
+  expect(
+    expectedFrequencies,
+    'Encountered an empty set of frequencies to assert on.  Is there something wrong with the test or the runner?'
+  ).to.have.length.greaterThan(0);
+  for (const [address, frequency] of Object.entries(expected_frequencies)) {
+    if (frequency === 0) {
+      expect(observed_frequencies).not.to.haveOwnProperty(address);
+    } else {
+      expect(observed_frequencies).to.haveOwnProperty(address).to.exist;
+      const actual_frequency = observed_frequencies[address];
+      const is_too_low = actual_frequency < frequency - tolerance;
+      expect(
```

Comment:
for the sake of readability here, since we aren't actually distinguishing too high and too low in the assertions, could we instead do ```if (isTooHigh || isTooLow) { expect.fail(`...tolerance error message...`) }```